### PR TITLE
Unify class-based tests

### PR DESCRIPTION
**Pull request by jorisdral**
Reduce duplication by using the unified class everywhere in our tests, which means we can remove the `Normal` and `Monoidal` classes altogether.

### CODE REVIEW

**Comment by jorisdral** [comment]
This is a WIP, so not yet ready for review

**Comment by jorisdral** [comment]
Ready for review now!

File: src/Database/LSMTree/Normal.hs [hunk 1]
```diff
@@ -332,7 +332,7 @@ lookups (Internal.NormalTable t) ks =
       Entry.Insert v            -> Found (Internal.deserialiseValue v)
       Entry.InsertWithBlob v br -> FoundWithBlob (Internal.deserialiseValue v)
                                                  (BlobRef br)
-      Entry.Mupdate _           -> error "Normal.lookups: unexpected Mupdate"
+      Entry.Mupdate v           -> Found (Internal.deserialiseValue v)
```
[the remark this answers is not in the document]
Comment (jorisdral) [reply src/Database/LSMTree/Normal.hs:335]: The `Normal` and `Monoidal` modules don't have tests targeting them anymore, so if we were to introduce an internal bug that converts an `Insert` to `Mupdate` for example, then we wouldn't see any failing tests, though users of the `Normal` module would start seeing the error. I tried to write some tests to check these errors don't happen, but they aren't very nice ones, or they're very verbose. So in the end I though maybe it's okay to just remove the error. `Mupdates` are equivalent to `Insert`s anyway in the `Normal` case

**Review by mheinzel — APPROVED** [review]

**Review by dcoutts — APPROVED** [review]
LGTM!

**Pull request merged**